packages/langchain/src/openai/chat.ts

```diff
@@ -0,0 +1,78 @@
+import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
+import { BaseMessage } from '@langchain/core/messages';
+import type { ChatResult } from '@langchain/core/outputs';
+import { AzureOpenAiChatClient as AzureOpenAiChatClientBase } from '@sap-ai-sdk/foundation-models';
+import { BaseChatModel } from '@langchain/core/language_models/chat_models';
+import { AzureOpenAiChatModel } from '@sap-ai-sdk/core';
+import { mapLangchainToAiClient, mapOutputToChatResult } from './util.js';
+import type {
+  AzureOpenAiChatCallOptions,
+  AzureOpenAiChatModelParams
+} from './types.js';
+
+/**
+ * LangChain chat client for Azure OpenAI consumption on SAP BTP.
+ */
+export class AzureOpenAiChatClient
+  extends BaseChatModel<AzureOpenAiChatCallOptions>
+  implements AzureOpenAiChatModelParams
+{
+  modelName: AzureOpenAiChatModel;
+  modelVersion?: string;
+  resourceGroup?: string;
```

Comment:
[minor] it's slightly awkward to have this mixed with the hyperparameters. Hypothetical, though somewhat unlikely case: OpenAI adds a hyperparameter called model version.

We could consider moving the hyperparameters out of the constructor to a separate method, e.g. 
```TS
new AzureOpenAiChatClient({ modelName: 'gpt-35-turbo' })
             .withConfig({ temperature: 0.5 });
```

However, probably not worth the refactoring effort for now, but maybe an idea for later

Reply:
This would not be aligned with all other langchain clients, where this is done in the constructor though